bin/mock_onfido.pl

```diff
@@ -0,0 +1,334 @@
+            created_at => Date::Utility->new()->datetime_iso8601,
+            name       => $req->{name},
+            status     => 'complete',
+            result     => 'clear',
+
+        };
+        $reports{$report_id} = $report;
+        push @reports, $report_id;
+    }
+    return [map { clone_and_remove_private($_) } @reports];
+}
+
+post '/v2/applicants/:applicant_id/checks' => sub {
+    my $c            = shift;
+    my $applicant_id = $c->stash('applicant_id');
+    my $check_id     = Data::UUID->new->create_str();
+    my $check        = {
+        id            => $check_id,
+        created_at    => Date::Utility->new()->datetime_iso8601,
+        href          => "/v2/applicants/$applicant_id/checks/$check_id",
+        type          => $c->param('type'),
+        status        => 'complete',
+        result        => 'clear',
+        redirect_uri  => 'https://somewhere.else',
+        results_uri   => "https://onfido.com/dashboard/information_requests/<REQUEST_ID>",
+        download_uri  => "https://onfido.com/dashboard/pdf/information_requests/<REQUEST_ID>",
+        reports       => create_report($c, $check_id),
+        tags          => $c->req->params->to_hash->{'tags[]'},
+        _applicant_id => $applicant_id,
+    };
+    $checks{$check_id} = $check;
+    return $c->render(json => clone_and_remove_private($check));
+};
+
+get '/v2/applicants/:applicant_id/checks/:check_id' => sub {
+    my $c            = shift;
+    my $applicant_id = $c->stash('applicant_id');
+    my $check_id     = $c->stash('check_id');
+    unless (exists($checks{$check_id}) && $checks{$check_id}{_applicant_id} eq $applicant_id) {
+        return $c->render(json => {status => 'Not Found'});
+    }
+    return $c->render(json => clone_and_remove_private($checks{$check_id}));
+};
+
+get '/v2/applicants/:applicant_id/checks' => sub {
+    my $c            = shift;
+    my $applicant_id = $c->stash('applicant_id');
+    my @checks =
+        sort { $b->{created_at} cmp $a->{created_at} }
+        map { clone_and_remove_private($_) } grep { $_->{_applicant_id} eq $applicant_id } values %checks;
+    return $c->render(json => {checks => \@checks});
+};
+
+get '/v2/checks/:check_id/reports' => sub {
+    my $c        = shift;
+    my $check_id = $c->stash('check_id');
+    my @reports =
+        sort { $b->{created_at} cmp $a->{created_at} }
+        map  { clone_and_remove_private($_) }
+        grep { $_->{_check_id} eq $check_id } values %reports;
+
+    return $c->render(json => {reports => \@reports});
+};
+
+get '/v2/checks/:check_id/reports/:report_id' => sub {
+    my $c         = shift;
+    my $check_id  = $c->stash('check_id');
+    my $report_id = $c->stash('report_id');
+    unless (exists($reports{$report_id}) && $reports{$report_id}{_check_id} eq $check_id) {
+        return $c->render(json => {status => 'Not Found'});
+    }
+    return $c->render(json => clone_and_remove_private($reports{$report_id}));
+};
+
+my @sdk_tokens;
```

Comment:
Not needed?

Reply:
It is used at line https://github.com/binary-com/perl-WebService-Async-Onfido/pull/14/files#diff-4c6dd007f53e659aa6603a19723f91a5R310 . Yes it only store token information but those information are not fetched again in the current version. I hope it will be used in other tests (bom-events, bom-rpc)